### PR TITLE
Patch so master to master

### PR DESCRIPTION
**Pull request by joefitter**

### CODE REVIEW

**Comment by joefitter** [comment]
@JoeChapman @easternbloc @danielepolencic @bendiggle @gxxm mind having a 👀 - patching so-master back to master. Contains the changes to the confirm controller and integration with hof-emailer

File: package.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
   },
   "dependencies": {
     "hmpo-form-wizard": "5.1.1",
+    "hof-emailer": "^1.0.0",
```
Comment (JoeChapman) [inline package.json:20]: there is a newer version of hof-emailer (1.1.0). Can this be updated?

File: test/lib/confirm-controller.js [hunk 2]
```diff
@@ -1,126 +1,409 @@
...
+        getValue: sinon.stub(),
+        conditionalTranslate: sinon.stub(),
+        getTranslation: sinon.stub()
+      };
+      ConfirmController = proxyquire('../../lib/confirm-controller', {
+        './base-controller': StubController,
+        './util/helpers': helpersStub,
+        'hof-emailer': EmailerStub
+      });
+      const steps = {
```
Comment (JoeChapman) [inline test/lib/confirm-controller.js:45]: isn't this another fixture? Most other fixtures in this PR have their own files

File: package.json (hunk 1 above)
Comment (joefitter) [reply package.json:20]: this is just bringing master up to date with so-master, so no changes have actually been made to this branch. I would suggest bumping hof-emailer in a separate PR

File: test/lib/confirm-controller.js (hunk 2 above)
Comment (joefitter) [reply test/lib/confirm-controller.js:45]: yeah I guess we could split it out, where do we draw the line though - do we add the field config to a separate file  (7 lines). The idea of fixtures were more real use case like scenarios, whereas these 'fixtures' are testing a simple use case

File: lib/util/helpers.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            return key;
+          }
+        }
+      } else {
+        path = `${base}.${key}.header`;
+        result = translate(path);
+        if (result === path) {
+          return key;
+        }
+      }
```
Comment (danielepolencic) [inline lib/util/helpers.js:140]: This is quite hard to follow. Is there any way we could rather write something like:

``` javascript
let result = [
  `${base}.${key}.summary`,
  `${base}.${key}.label`,
  `${base}.${key}.legend`,
  `${base}.${key}.header`
].some(path => {
  return translate(path) === path;
});
return result ? key : undefined;
```

File: test/lib/confirm-controller.js [hunk 4]
```diff
@@ -1,126 +1,409 @@
...
-      'field-four': 4
-    };
-
-    controller.options = {
-      steps: {
-        '/one': {
-          fields: ['field-one', 'field-two']
+    StubController.prototype.locals = sinon.stub().returns({});
+    StubController.prototype.get = sinon.stub();
+    EmailerStub.prototype.sendEmails = sinon.stub().returns(new Promise(resolve => resolve()));
```
Comment (danielepolencic) [inline test/lib/confirm-controller.js:22]: `new Promise(resolve => resolve())` is `Promise.resolve()`

File: package.json (hunk 1 above)
Comment (JoeChapman) [reply package.json:20]: ok

File: lib/util/helpers.js (hunk 3 above)
Comment (joefitter) [reply lib/util/helpers.js:140]: that's much nicer

File: test/lib/confirm-controller.js (hunk 4 above)
Comment (joefitter) [reply test/lib/confirm-controller.js:22]: magic!

File: lib/util/helpers.js (hunk 3 above)
Comment (danielepolencic) [reply lib/util/helpers.js:140]: I haven't tested the code though.

Comment (joefitter) [reply lib/util/helpers.js:140]: @danielepolencic I haven't updated this yet as up to my eyeballs in project work, but there is a potentially nicer solution used elsewhere in the architecture - i18n-lookup (one of Lenny's packages) - pass in an array of keys and it returns the first one found

**Comment by joefitter** [comment]
@JoeChapman @easternbloc @danielepolencic @bendiggle mind having a look over this again please? @danielepolencic - changes made as suggested. This now also contains some more changes mainly to the confirm controller

**Comment by joefitter** [comment]
ignore - rebase fail. Will resubmit!
